docs/contributor/ui/README.md

```diff
@@ -0,0 +1,185 @@
+
+[Official List Documentation](https://github.com/kyma-project/busola/blob/main/docs/extensibility/20-list-columns.md)
+
+[Official List and Detail Widgets](https://github.com/kyma-project/busola/blob/main/docs/extensibility/50-list-and-details-widgets.md)
+
+### Detail
+```yaml
+body:
+    - name: configuration
+      widget: Panel
+      source: spec
+      children:
+        - name: spec.fileShareName
+          source: fileShareName
+          widget: Labels
+        - name: spec.capacityGb
+          source: capacityGb
+          widget: Labels
+        - name: spec.location
+          source: location
+          widget: Labels
+        - name: spec.tier
+          source: tier
+          widget: Labels
+    - name: status
+      widget: Panel
+      source: status
+      children:
+        - widget: Labels
+          source: state
+          name: status.state
+```
+
+`widget` refers to the component interfaces built into Busola. See link below for official documentation on widgets.
+
+`children` are an array of child widgets. Note, the `source` in each child is relative to its parent.
+
+[Official Detail Documentation](https://github.com/kyma-project/busola/blob/main/docs/extensibility/30-details-summary.md)
+
+[Official List and Detail Widgets](https://github.com/kyma-project/busola/blob/main/docs/extensibility/50-list-and-details-widgets.md)
+
+### Form
+```yaml
+- path: spec.capacityGb
+  simple: true
+  name: spec.capacityGb
+  required: true
+- path: spec.fileShareName
+  simple: true
+  name: spec.fileShareName
+  required: true
+- path: spec.location
+  simple: true
+  name: spec.location
+  required: true
+- path: spec.tier
+  simple: true
+  name: spec.tier
+  required: true
+```
+
+`simple` is a boolean used to display the field in the simple fom. By default, it is `false`
+
+[Official Forms Documentation](https://github.com/kyma-project/busola/blob/main/docs/extensibility/40-form-fields.md)
+
+<a id="translations"></a>
+### Translations
+```yaml
+en:
+  spec.tier: Tier
+  spec.location: Location
+  spec.fileShareName: File Share Name
+  spec.capacityGb: Capacity (Gb)
+  spec.ipRange: IP Range
+  configuration: Configuration
+  status.state: State
+  status: Status
+```
+
+`translations` is optional, but it supports languages formatted for [i18next](https://www.i18next.com/). Translations prettify the `name` field of the aformentioned sections.
+They are key-value pairs.
+
+
+[Official Translations Documentation](https://github.com/kyma-project/busola/blob/main/docs/extensibility/translations-section.md)
+
+
+# Generating the ConfigMap
+
+The generation of ConfigMap is handled through `kustomize`. Kustomize  combines general, list, detail, form, and translations
+into a single ConfigMap.
+
+Because each CustomResource will have its own UI that must be generated, there is a single macro in the `Makefile` to generate all the UI.
```

Comment:
```suggestion
Because each CustomResource has its own UI that must be generated, there is a single macro in the `Makefile` to generate all the UIs.
```